templates/secrets-template.yml

```diff
@@ -165,6 +153,7 @@ objects:
     kubeconfig: ${KUBE_CONFIG}
     image-pull.dockerconfigjson: "${IMAGE_PULL_DOCKER_CONFIG}"
     gcp.api-credentials: "${GCP_API_CREDENTIALS}"
+    dataplane-observability-config.yaml: ${DATAPLANE_OBSERVABILITY_CONFIG}
```

Comment:
I wonder if this should become its own K8s secret

Reply:
After talking through other channels we decided to keep it in the secret used in the PR for now.
In the future we might change it.